implementation/src/main/java/io/smallrye/mutiny/operators/uni/builders/UniJoinFirst.java

```diff
@@ -96,6 +119,11 @@ private void onFailure(Throwable failure) {
                         if (failures.size() == unis.size()) {
                             cancelled.set(true);
                             subscriber.onFailure(new CompositeException(failures));
+                        } else if (concurrency != -1) {
+                            int nextIndex = nextSubscriptionIndex.incrementAndGet();
+                            if (nextIndex < unis.size()) {
```

Comment:
Same comment here - we need to check for cancellation before doing the subscription.

Reply:
@jponge it needs to check for cancellation too.